## Introducción <a id='intro'></a>
Este proyecto tiene como objetivo analizar los datos de transmisión de música online para probar una hipótesis sobre las preferencias musicales de los usuarios en dos ciudades, Springfield y Shelbyville. A través de tres etapas (descripción de los datos, preprocesamiento y prueba de hipótesis), se comparará el comportamiento de los usuarios en ambas ciudades utilizando métodos estadísticos para validar o rechazar la hipótesis planteada. Al final, se espera extraer conclusiones valiosas que ayuden a tomar decisiones basadas en los datos analizados.

### Objetivo:
Prueba la hipótesis:
1. La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### Etapas
Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.








# Contenido <a id='back'></a>

* [Introducción](#intro)
* [Etapa 1. Descripción de los datos](#data_review)
    * [Conclusiones](#data_review_conclusions)
* [Etapa 2. Preprocesamiento de datos](#data_preprocessing)
    * [2.1 Estilo del encabezado](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusiones](#data_preprocessing_conclusions)
* [Etapa 3. Prueba de hipótesis](#hypothesis)
    * [3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades](#activity)
* [Conclusiones](#end)

## Etapa 1. Descripción de los datos <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás `pandas`, así que impórtalo.

In [2]:
# Importar pandas

import pandas as pd


Lee el archivo `music_project_en.csv` de la carpeta `/datasets/` y guárdalo en la variable `df`:

In [3]:
# Leer el archivo y almacenarlo en df
df = pd.read_csv("/datasets/music_project_en.csv")


Muestra las 10 primeras filas de la tabla:

In [4]:
# Obtener las 10 primeras filas de la tabla df
print(df.head(10))

     userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [5]:
# Obtener la información general sobre nuestros datos
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB
None


Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: `object`.

Según la documentación:
- `' userID'`: identificador del usuario o la usuaria;
- `'Track'`: título de la canción;
- `'artist'`: nombre del artista;
- `'genre'`: género de la pista;
- `'City'`: ciudad del usuario o la usuaria;
- `'time'`: la hora exacta en la que se reprodujo la canción;
- `'Day'`: día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:
1. Algunos encabezados están en mayúsculas, otros en minúsculas.
2. Hay espacios en algunos encabezados.
3. `Detecta el tercer problema por tu cuenta y descríbelo aquí`.
*Respuesta* -> Otro problema que presentan algunos nombres de las columnas es que tienen nombres confusos. Hay un caso en que el nombre está constituido por dos palabras, `' userID'`, pero no posee un objeto separador entre ellas (posible arreglo utilizando snake_case). También se puede señalar que la columna `'time'` es poco descriptiva, por lo que se debería cambiar a un nombre que transmita mayor claridad de su significado.




### Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles: <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`

El Data frame contiene filas que muestran cada vez que un usuario reprodujo una canción en un determinado periodo de tiempo. Cada fila está compuesta por varios atributos o características señaladas en las columnas que componen los datos, dentro de las cuales encontramos información como el título de la canción que se reprodujo, el nombre del artista, el género musical de la canción, que usuario la reprodujo, cuando (día y hora) y donde (ciudad).


`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`

Actualmente a simple vista hay suficientes datos para proporcionar una respuesta a la hipótesis "La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad", ya que tenemos datos que indican la actividad de usuarios de ambas ciudades y también por día de la semana. Aunque cabe destacar que esto podría cambiar en el proceso de limpieza y análisis de los datos, ya que aún no se revisa la calidad de los datos y si estos tienen una muestra similar para cada ciudad y cada día.


`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`

Con la información proporcionada al usar la función info. podemos visualizar que tres columnas contienen actualmente valores ausentes(__Track__, __artist__ y __genre__), ya que su sumatoria de filas en la columna `Non-null Count` es inferior que el número total de filas del Data frame. Además se puede ver que todas las columnas han sido categorizadas como tipos de dato `object` cuando la columna __time__ debería estar como tipo de dato `float64`.

## Etapa 2. Preprocesamiento de datos <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.


### Estilo del encabezado <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [6]:
# Muestra los nombres de las columnas
print(df.columns)


Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')


Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

In [7]:
# Bucle en los encabezados poniendo todo en minúsculas
new_col_names = []

for old_name in df.columns:
    name_lowered = old_name.lower()
    new_col_names.append(name_lowered)

df.columns = new_col_names

print(df.columns)

Index(['  userid', 'track', 'artist', 'genre', '  city  ', 'time', 'day'], dtype='object')


Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

In [8]:
# Bucle en los encabezados eliminando los espacios
new_col_names = []

for old_name in df.columns:
    name_stripped = old_name.strip()
    new_col_names.append(name_stripped)

df.columns = new_col_names

print(df.columns)
 


Index(['userid', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [9]:
# Cambiar el nombre de la columna "userid"

df.rename(columns = {'userid':'user_id'}, inplace = True)




Comprueba el resultado. Muestra los encabezados una vez más:

In [10]:
# Comprobar el resultado: la lista de encabezados
print(df.columns)


Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


### Valores ausentes <a id='missing_values'></a>
 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [11]:
# Calcular el número de valores ausentes
print(df.isna().sum())

user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64


No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [12]:
# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'
columns_to_replace = ['track', 'artist', 'genre']
for col in columns_to_replace:
    df[col].fillna('unknown', inplace=True)

Ahora comprueba el resultado para asegurarte de que después del reemplazo no haya valores ausentes en el conjunto de datos. Para hacer esto, cuenta los valores ausentes nuevamente.

In [13]:
# Contar valores ausentes
print(df.isna().sum())

user_id    0
track      0
artist     0
genre      0
city       0
time       0
day        0
dtype: int64


### Duplicados <a id='duplicates'></a>
Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [14]:
# Contar duplicados explícitos
print(df.duplicated().sum())

3826


Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [15]:
# Eliminar duplicados explícitos
df = df.drop_duplicates().reset_index(drop=True)



Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [16]:
# Comprobar de nuevo si hay duplicados

print(df.duplicated().sum())


0


Ahora queremos deshacernos de los duplicados implícitos en la columna `genre`. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:
* Extrae la columna `genre` del DataFrame.
* Llama al método que devolverá todos los valores únicos en la columna extraída.


In [17]:
# Inspeccionar los nombres de géneros únicos

print(df["genre"].sort_values().unique())


['acid' 'acoustic' 'action' 'adult' 'africa' 'afrikaans' 'alternative'
 'ambient' 'americana' 'animated' 'anime' 'arabesk' 'arabic' 'arena'
 'argentinetango' 'art' 'audiobook' 'avantgarde' 'axé' 'baile' 'balkan'
 'beats' 'bigroom' 'black' 'bluegrass' 'blues' 'bollywood' 'bossa'
 'brazilian' 'breakbeat' 'breaks' 'broadway' 'cantautori' 'cantopop'
 'canzone' 'caribbean' 'caucasian' 'celtic' 'chamber' 'children' 'chill'
 'chinese' 'choral' 'christian' 'christmas' 'classical' 'classicmetal'
 'club' 'colombian' 'comedy' 'conjazz' 'contemporary' 'country' 'cuban'
 'dance' 'dancehall' 'dancepop' 'dark' 'death' 'deep' 'deutschrock'
 'deutschspr' 'dirty' 'disco' 'dnb' 'documentary' 'downbeat' 'downtempo'
 'drum' 'dub' 'dubstep' 'eastern' 'easy' 'electronic' 'electropop' 'emo'
 'entehno' 'epicmetal' 'estrada' 'ethnic' 'eurofolk' 'european'
 'experimental' 'extrememetal' 'fado' 'film' 'fitness' 'flamenco' 'folk'
 'folklore' 'folkmetal' 'folkrock' 'folktronica' 'forró' 'frankreich'
 'französisch' 

Busca en la lista para encontrar duplicados implícitos del género `hiphop`. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para deshacerte de ellos, crea una función llamada `replace_wrong_genres()` con dos parámetros:
* `wrong_genres=`: esta es una lista que contiene todos los valores que necesitas reemplazar.
* `correct_genre=`: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna `'genre'` de la tabla `df`, es decir, remplazar cada valor de la lista `wrong_genres` por el valor en `correct_genre`.

Dentro del cuerpo de la función, utiliza un bucle `'for'` para iterar sobre la lista de géneros incorrectos, extrae la columna `'genre'` y aplica el método `replace` para hacer correcciones.

In [18]:
# Función para reemplazar duplicados implícitos
def replace_wrong_values(df,columns,wrong_genres, correct_genre):
    for wrong_genre in wrong_genres:
        df['genre'] = df['genre'].replace(wrong_genre,correct_genre)
    return df




Ahora, llama a `replace_wrong_genres()` y pásale tales argumentos para que retire los duplicados implícitos (`hip`, `hop` y `hip-hop`) y los reemplace por `hiphop`:

In [19]:
# Eliminar duplicados implícitos
duplicates = ["hip", "hop", "hip-hop"]
name = "hiphop"

df = replace_wrong_values(df,"genre",duplicates,name)

Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna `'genre'` una vez más:

In [20]:
# Comprobación de duplicados implícitos

print(df["genre"].sort_values().unique())

['acid' 'acoustic' 'action' 'adult' 'africa' 'afrikaans' 'alternative'
 'ambient' 'americana' 'animated' 'anime' 'arabesk' 'arabic' 'arena'
 'argentinetango' 'art' 'audiobook' 'avantgarde' 'axé' 'baile' 'balkan'
 'beats' 'bigroom' 'black' 'bluegrass' 'blues' 'bollywood' 'bossa'
 'brazilian' 'breakbeat' 'breaks' 'broadway' 'cantautori' 'cantopop'
 'canzone' 'caribbean' 'caucasian' 'celtic' 'chamber' 'children' 'chill'
 'chinese' 'choral' 'christian' 'christmas' 'classical' 'classicmetal'
 'club' 'colombian' 'comedy' 'conjazz' 'contemporary' 'country' 'cuban'
 'dance' 'dancehall' 'dancepop' 'dark' 'death' 'deep' 'deutschrock'
 'deutschspr' 'dirty' 'disco' 'dnb' 'documentary' 'downbeat' 'downtempo'
 'drum' 'dub' 'dubstep' 'eastern' 'easy' 'electronic' 'electropop' 'emo'
 'entehno' 'epicmetal' 'estrada' 'ethnic' 'eurofolk' 'european'
 'experimental' 'extrememetal' 'fado' 'film' 'fitness' 'flamenco' 'folk'
 'folklore' 'folkmetal' 'folkrock' 'folktronica' 'forró' 'frankreich'
 'französisch' 

### Tus observaciones <a id='data_preprocessing_conclusions'></a>

`Describe brevemente lo que has notado al analizar duplicados, cómo abordaste sus eliminaciones y qué resultados obtuviste.`

Al analizar la columna `genre` en búsqueda de duplicados se puede observar que existen muchos valores únicos, entre los cuales hay varios errores de tipeo y también se nombran muchos subgéneros musicales. Si bien esta columna nos podría dar insights con respecto a los gustos musicales de cada ciudad o determinados patrones según el día en que se escucha música, primero tendría que pasar por un proceso de limpieza y ordenamiento de los datos.

## Etapa 3. Prueba de hipótesis <a id='hypothesis'></a>

### Hipótesis: comparar el comportamiento del usuario o la usuaria en las dos ciudades <a id='activity'></a>

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

* Agrupa a los usuarios y las usuarias por ciudad.
* Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final:
`df.groupby(by='....')['column'].method()`Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.



In [21]:
# Contar las canciones reproducidas en cada ciudad
print(df.groupby(by = "city")["track"].count())

print(df.groupby(by = "city")["user_id"].nunique())


city
Shelbyville    18512
Springfield    42741
Name: track, dtype: int64
city
Shelbyville    12423
Springfield    29358
Name: user_id, dtype: int64


`Comenta tus observaciones aquí`

Se puede visualizar que Springfield compone más de dos tercios de las filas de la data, por lo que podríamos suponer que más habitantes de dicha ciudad escuchan música en comparación con los de Shelbyville, sin embargo, también estos números podrían ser causados por duplicados de usuarios, por lo que otra posibilidad sería que ambas ciudades tengan una cantidad de muestra de usuarios parecida, pero que los de Springfield hayan escuchado un mayor número de canciones. Para salir de la duda también añadí una fórmula extra para ver los valores únicos de la columna `user_id` por ciudad, en la que se puede observar que si bien en ambas ciudades hay usuarios que escucharon más de una canción, los valores únicos de Springfield siguen componiendo un 70% del total de las filas.


Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.


In [22]:
# Calcular las canciones reproducidas en cada uno de los tres días
print(df.groupby(by = "day")["track"].count())


print(df.groupby(by = "day")["user_id"].nunique())


day
Friday       21840
Monday       21354
Wednesday    18059
Name: track, dtype: int64
day
Friday       16636
Monday       16144
Wednesday    13614
Name: user_id, dtype: int64


`Comenta tus observaciones aquí`

En el caso de la cantidad de canciones reproducidas por día, se puede observar una clara alza los días lunes y viernes, mientras que el miércoles es el día con menos canciones reproducidas. Esta tendencia también se observa en el número de usuarios que escucharon canciones por día. 

Una opción interesante seria ver si es que los géneros musicales escuchados al inicio de la semana varían mucho con respecto a los escuchados al final antes del fin de semana y si estas tendencias ocurren en ambas ciudades.

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función `number_tracks()` para calcular el número de canciones reproducidas en un determinado día **y** ciudad. La función debe aceptar dos parámetros:

- `day`: un día de la semana para filtrar. Por ejemplo, `'Monday'` (lunes).
- `city`: una ciudad para filtrar. Por ejemplo, `'Springfield'`.

Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [33]:
# Declara la función number_tracks() con dos parámetros: day= y city=.

def number_tracks(day, city):
    
    # Almacena las filas del DataFrame donde el valor en la columna 'day' es igual al parámetro day=
    
    filtered_result_1 = df[df['day'] == day]
    
     # Filtra las filas donde el valor en la columna 'city' es igual al parámetro city=
    
    filtered_result_2 = filtered_result_1[filtered_result_1['city'] == city]
    
    return filtered_result_2["user_id"].count()



# Extrae la columna 'user_id' de la tabla filtrada y aplica el método count()
 # Devolve el número de valores de la columna 'user_id'

print(number_tracks('Monday','Springfield' ))


    

15740


Llama a `number_tracks()` seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [34]:
# El número de canciones reproducidas en Springfield el lunes
print(number_tracks('Monday','Springfield' ))

15740


In [35]:
# El número de canciones reproducidas en Shelbyville el lunes
print(number_tracks('Monday','Shelbyville' ))


5614


In [29]:
# El número de canciones reproducidas en Springfield el miércoles
print(number_tracks('Wednesday','Springfield' ))


11056


In [30]:
# El número de canciones reproducidas en Shelbyville el miércoles
print(number_tracks('Wednesday','Shelbyville' ))


7003


In [31]:
# El número de canciones reproducidas en Springfield el viernes
print(number_tracks('Friday','Springfield' ))


15945


In [32]:
# El número de canciones reproducidas en Shelbyville el viernes

print(number_tracks('Friday','Shelbyville' ))


5895


# Conclusiones <a id='end'></a>



`Comenta si la hipótesis es correcta o se debe rechazar. Explica tu razonamiento.`

La hipótesis "La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad" puede ser aceptada pero solo parcialmente. Si bien la última información recopilada acerca de la cantidad de usuarios si nos da indicios que existe una diferencia en la actividad de los usuarios según el día y la ciudad, ya que:

           L   |   M   |   V
Springfield - 15740 - 11056 - 15945

Shelbyville -  5614 -  7003 -  5895

Se puede ver que al contrario de Springfield, los habitantes de Shelvyville prefieren escuchar canciones el miercoles vs los otros dos días, lo que efectivamente contaría como una diferencia, sin embargo, existe mucha ambiguedad con respecto a los datos utilizados, ya que no sabemos la cantidad de días que los componen, si es que los usuarios analizados de ambas ciudades poseen caracteristicas similares que los hagan comparables (ej.edad, genero) o que los datos esten bien distribuidos lograr para ser una muestra representativa de la ciudad analizada, etc...
